### ISSUE TITLE
Docker environment for testing cross-chain orchestration

### ISSUE THREAD

**Issue by turadg** [Agoric/Agoric#8896]
As a contract developer, I need a docker environment with simulated Osmosis, Cosmos, Agoric, IBC Relayers, and test tokens.
* Candidates: [interchaintest](https://github.com/strangelove-ventures/interchaintest) (ivan has experience) + [Starship](https://github.com/cosmology-tech/starship/ 
) (patrick demoed)
* Considerations:
  * IBC versions (we need IBC v6, most docs geared at v7)
  * we need contracts/state on other chains (needs swap contracts + assets)
  * faucet for tokens, or mnemonics preloaded with funds.

```[tasklist]
### Dependencies
- [ ] https://github.com/Agoric/agoric-sdk/issues/8895
```

**Comment by LuqiPan** [comment]
As discussed in the project management meeting, @0xpatrickdev will come back with a list of things that are missing from Agoric chain at genesis that need to be added to starship and then we'll regroup and find the owners to address those missing things

**Comment by dckc** [comment]
@jovonni and @ivanlei noted another candidate:

 - [osmosis\-labs/test\-tube: Test tube for cosmos\-sdk chains integration test, written in Rust\!](https://github.com/osmosis-labs/test-tube)

**Comment by dckc** [comment]
@0xpatrickdev notes his WIP: [QUICKSTART](https://github.com/0xpatrickdev/starship-demo/pull/2/files#diff-147873af8be465c9803fc2026de7840536a18f4e37045ae94bd10ce4e77810d3)
in https://github.com/0xpatrickdev/starship-demo
https://github.com/0xpatrickdev/starship-demo/pull/2

**Comment by 0xpatrickdev** [comment]
[starship-demo#700a5f68](https://github.com/0xpatrickdev/starship-demo/tree/07e7ddff1e7ca79b311463ab3f18a2f72bf2d208) contains changes implemented by the cosmology team to integrate an agoric chain image via agoric-3-proposals. The chain starts successfully, but from a new genesis state instead of the simulated state from agoric-3-proposals.

In [starship-demo repo#07e7ddf](https://github.com/0xpatrickdev/starship-demo/tree/07e7ddff1e7ca79b311463ab3f18a2f72bf2d208), the latest head, the agoric configuration implements the custom [scripts directives](https://docs.cosmology.zone/starship/config/chains#scripts-optional) available for building the chain. Presumably, these could be modified to preserve the existing genesis state and send tokens to the starship `genesis` account to allow it validate. So far my attempts at this have been unsuccessful, but I can share my attempts which may give someone else a head start.

An alternative approach might involve executing core proposals on top of the new genesis state. I explored this path today (init-network.js, init-localchain.js, init-orchestration.js) but faced some headwinds:
```
2024-05-14T19:11:32.928Z SwingSet: xsnap: v17: UnhandledPromiseRejectionWarning: (RangeError#1)
2024-05-14T19:11:32.937Z SwingSet: xsnap: v1: RemoteError(error:liveSlots:v2#70003)#3 ERROR_NOTE: Sent as error:liveSlots:v1#70002
2024-05-14T19:11:32.940Z SwingSet: xsnap: v10: UnhandledPromiseRejectionWarning: (RemoteError(error:liveSlots:v1#70002)#2)
2024-05-14T19:11:32.940Z SwingSet: xsnap: v10: RemoteError(error:liveSlots:v1#70002)#2: Vat Creation Error: Error: RangeError: Expected (an undefined) is same as (a string)
2024-05-14T19:11:32.940Z SwingSet: xsnap: v10: Error: Vat Creation Error: Error: RangeError: Expected (an undefined) is same as (a string)
```

**Comment by dckc** [comment]
> ... but faced some headwinds:
> 
> ```
> 2024-05-14T19:11:32.940Z SwingSet: xsnap: v10: RemoteError(error:liveSlots:v1#70002)#2: Vat Creation Error: Error: RangeError: Expected (an undefined) is same as (a string)
> ```

@0xpatrickdev and I managed to get just a little more info: `Expected undefined is same as Interface` at which point I recognized this as symptoms of building bundles with an incompatible endo version (#8826).

**Comment by 0xpatrickdev** [comment]
Here is a comparison of vats available in [starship-demo repo#07e7ddf](https://github.com/0xpatrickdev/starship-demo/tree/07e7ddff1e7ca79b311463ab3f18a2f72bf2d208) versus a normal a3p: 

<details>
<summary>normal</summary>

```
# ps aux | grep xsnap | awk '{print $12}' | sort -t: -k1.2n
xsnap
v1:bootstrap
v2:vatAdmin
v4:vattp
v5:timer
v6:agoricNames
v7:board
v8:priceAuthority
v9:zoe
v10:bridge
v11:provisioning
v13:zcf-mintHolder-BLD
v14:bank
v15:zcf-b1-2371d-econCommitteeCharter
v16:zcf-mintHolder-ATOM
v17:zcf-mintHolder-USDC_axl
v18:zcf-mintHolder-USDC_grv
v19:zcf-mintHolder-USDT_axl
v20:zcf-mintHolder-USDT_grv
v21:zcf-mintHolder-DAI_axl
v22:zcf-mintHolder-DAI_grv
v23:zcf-b1-e9e0b-feeDistributor
v24:zcf-b1-941ef-economicCommittee
v25:zcf-b1-9f877-provisionPool-governor
v26:zcf-b1-9f877-ATOM-USD_price_feed-governor
v36:zcf-b1-92c07-reserve
v45:zcf-b1-a5683-auctioneer
v48:zcf-b1-6c08a-vaultFactory
v55:zcf-mintHolder-stATOM
v57:zcf-b1-4522b-stATOM-USD_price_feed
v58:zcf-b1-0b217-scaledPriceAuthority-stATOM
```
</details>

<details>
<summary>starship</summary>

```
xsnap
v1:bootstrap
v2:vatAdmin
v4:vattp
v5:timer
v6:agoricNames
v7:board
v8:priceAuthority
v9:zoe
v10:bridge
v11:provisioning
v13:zcf-mintHolder-BLD
v14:bank
```
</details>

The diff is only v15 onwards, which only seems to be inter economy dependencies. I don't think these are completely necessary, and was surprised to see some behavior like smart wallets not working as expected. Provisioning a smart wallet results in unexpected behavior. When attempting to first add 10_000_000uist to the provision pool addr, `agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346`, I am met with the same results. In vstorage:

```
egress: Failed to parse: {"nickname":"my-wallet","peer":"agoric15xddzse9lq74cyt6ev9d7wywxerenxdgxsdc3m","powerFlags":["SMART_WALLET"]}
```

<details>
<summary>vstorage screenshot</summary>

<img width="902" alt="Screenshot 2024-05-15 at 6 06 56 PM" src="https://github.com/Agoric/agoric-sdk/assets/11021913/2a72e6a7-d1b6-4f29-8501-e2e351deea98">
</details>

_Still figuring out how to set `DEBUG: "SwingSet:ls,SwingSet:vat"` for meaningful swingset logs in this env._

Here is the beginning section of the validator logs:

<details>
<summary>starship validator logs</summary>

```
10:30PM INF agd delegating to JS executable args=["ag-chain-cosmos","--home","/root/.agoric","start"] binary=/usr/src/agoric-sdk/packages/cosmic-swingset/src/entrypoint.js
Removing intrinsics.Symbol.dispose
Removing intrinsics.Symbol.asyncDispose
10:30PM INF starting node with ABCI Tendermint in-process
10:30PM INF service start impl=multiAppConn module=proxy msg={}
10:30PM INF service start connection=query impl=committingClient module=abci-client msg={}
10:30PM INF service start connection=snapshot impl=committingClient module=abci-client msg={}
10:30PM INF service start connection=mempool impl=committingClient module=abci-client msg={}
10:30PM INF service start connection=consensus impl=committingClient module=abci-client msg={}
10:30PM INF service start impl=EventBus module=events msg={}
10:30PM INF service start impl=PubSub module=pubsub msg={}
10:30PM INF service start impl=IndexerService module=txindex msg={}
10:30PM INF ABCI Handshake App Info hash= height=0 module=consensus protocol-version=0 software-version=0.35.0-u14.1
10:30PM INF ABCI Replay Blocks appHeight=0 module=consensus stateHeight=0 storeHeight=0
10:30PM INF initializing blockchain state from genesis.json
10:30PM INF created new capability module=ibc name=ports/transfer
10:30PM INF port binded module=x/ibc/port port=transfer
10:30PM INF claimed capability capability=1 module=transfer name=ports/transfer
10:30PM INF created new capability module=ibc name=ports/icahost
10:30PM INF port binded module=x/ibc/port port=icahost
10:30PM INF claimed capability capability=2 module=icahost name=ports/icahost
Loading slog sender modules: @agoric/telemetry/src/slog-file.js @agoric/telemetry/src/flight-recorder.js
2024-05-15T22:30:26.882Z launch-chain: Launching SwingSet kernel
2024-05-15T22:31:08.152Z launch-chain: Launched SwingSet kernel
2024-05-15T22:31:08.154Z block-manager: block bootstrap
10:31PM INF Completed ABCI Handshake - CometBFT and App are synced appHash="���B��\x1c\x14���șo�$'�A�d��L���\x1bxR�U" appHeight=0 module=consensus
10:31PM INF Version info abci=0.17.0 block=11 cmtbft_version=0.34.30 commit_hash= p2p=8
10:31PM INF This node is a validator addr=3F1738D7F374F8CFD2DFFCD596B5C3B57BBD9513 module=consensus pubKey=N95XLNvWYako/TGXKS4A92GZHxKdHp7UslnUDNjJwQE=
10:31PM INF P2P Node ID ID=79142527a2677ace2d32ad30876806326b09a372 file=/root/.agoric/config/node_key.json module=p2p
10:31PM INF Adding persistent peers addrs=[] module=p2p
10:31PM INF Adding unconditional peer ids ids=[] module=p2p
10:31PM INF Add our address to book addr={"id":"79142527a2677ace2d32ad30876806326b09a372","ip":"0.0.0.0","port":26656} book=/root/.agoric/config/addrbook.json module=p2p
10:31PM INF service start impl=Node msg={}
10:31PM INF Starting pprof server laddr=localhost:6060
10:31PM INF service start impl="P2P Switch" module=p2p msg={}
10:31PM INF service start impl=BlockchainReactor module=blockchain msg={}
10:31PM INF service start impl=ConsensusReactor module=consensus msg={}
10:31PM INF Reactor  module=consensus waitSync=false
10:31PM INF service start impl=ConsensusState module=consensus msg={}
10:31PM INF serve module=rpc-server msg={}
10:31PM INF service start impl=baseWAL module=consensus msg={} wal=/root/.agoric/data/cs.wal/wal
10:31PM INF service start impl=Group module=consensus msg={} wal=/root/.agoric/data/cs.wal/wal
10:31PM INF service start impl=TimeoutTicker module=consensus msg={}
10:31PM INF Searching for height height=1 max=0 min=0 module=consensus wal=/root/.agoric/data/cs.wal/wal
10:31PM INF Searching for height height=0 max=0 min=0 module=consensus wal=/root/.agoric/data/cs.wal/wal
10:31PM INF Found height=0 index=0 module=consensus wal=/root/.agoric/data/cs.wal/wal
10:31PM INF Catchup by replaying consensus messages height=1 module=consensus
10:31PM INF Replay: Done module=consensus
10:31PM INF service start impl=Evidence module=evidence msg={}
10:31PM INF service start impl=StateSync module=statesync msg={}
10:31PM INF service start impl=PEX module=pex msg={}
10:31PM INF service start book=/root/.agoric/config/addrbook.json impl=AddrBook module=p2p msg={}
10:31PM INF Saving AddrBook to file book=/root/.agoric/config/addrbook.json module=p2p size=0
10:31PM INF Ensure peers module=pex numDialing=0 numInPeers=0 numOutPeers=0 numToDial=10
10:31PM INF No addresses to dial. Falling back to seeds module=pex
10:31PM INF serve module=api-server msg={}
10:31PM INF Timed out dur=792.698083 height=1 module=consensus round=0 step=1
10:31PM INF received proposal module=consensus proposal={"Type":32,"block_id":{"hash":"4899148FD09F715727324088E3DA144417E2A080AFFB5815F1A5FE987886F56E","parts":{"hash":"DF68A321CF48246BFEE2C669A4B89A9B8CE12F9FF0010305345D16207CAC96BC","total":1}},"height":1,"pol_round":-1,"round":0,"signature":"nEG9NW/eGC3buYUkQfA7wj59Lt3JmtJCr6ssU+4kQQnv0HBdPziBWmzjw5IJGVYr+8t+hv6Tf931Ljp0A/pbDw==","timestamp":"2024-05-15T22:31:21.80795234Z"} proposer=3F1738D7F374F8CFD2DFFCD596B5C3B57BBD9513
10:31PM INF received complete proposal block hash=4899148FD09F715727324088E3DA144417E2A080AFFB5815F1A5FE987886F56E height=1 module=consensus
10:31PM INF finalizing commit of block hash={} height=1 module=consensus num_txs=0 root=E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855
10:31PM INF minted coins from module account amount=8238904ubld from=mint module=x/bank
2024-05-15T22:31:21.849Z block-manager: block 1 begin
10:31PM INF executed block height=1 module=state num_invalid_txs=0 num_valid_txs=0
2024-05-15T22:31:21.869Z block-manager: block 1 commit
10:31PM INF Timed out dur=400 height=1 module=consensus round=0 step=3
10:31PM INF commit synced commit=436F6D6D697449447B5B323532203234302031323820323620323530203234352031333220323337203430203135362032203134362031313320323320323034203133342031353520323438203234302031393320353320313633203231372034312031393720393220313831203232392032323020382033332035345D3A317D
10:31PM INF committed state app_hash=FCF0801AFAF584ED289C02927117CC869BF8F0C135A3D929C55CB5E5DC082136 height=1 module=state num_txs=0
10:31PM INF indexed block events height=1 module=txindex
10:31PM INF Timed out dur=326.159666 height=2 module=consensus round=0 step=1
10:31PM INF received proposal module=consensus proposal={"Type":32,"block_id":{"hash":"0D3208AD5FA90D1C69BF8A91E6EECA8EB4E7099CF33A4A29B218233AF5A23C98","parts":{"hash":"C533C5CEB827EAD492CB29C577638D89B32E547F36F8AF7B91060541C914FCC4","total":1}},"height":2,"pol_round":-1,"round":0,"signature":"9pfINSDO8cxkaJCHmW3rRsOPWbTeABpqpak/lf7+1W4mmuXwaw9bU6O4o6SC4UgK/ZtahZ61Tles+ynEDJ4mCw==","timestamp":"2024-05-15T22:31:22.688775883Z"} proposer=3F1738D7F374F8CFD2DFFCD596B5C3B57BBD9513
10:31PM INF received complete proposal block hash=0D3208AD5FA90D1C69BF8A91E6EECA8EB4E7099CF33A4A29B218233AF5A23C98 height=2 module=consensus
10:31PM INF finalizing commit of block hash={} height=2 module=consensus num_txs=0 root=FCF0801AFAF584ED289C02927117CC869BF8F0C135A3D929C55CB5E5DC082136
10:31PM INF minted coins from module account amount=8238905ubld from=mint module=x/bank
2024-05-15T22:31:22.806Z block-manager: block 2 begin
```
</details>

It seems like Cosmosvisor is being used , but I'm not seeing the [repoconfig.sh: No such file or directory error](https://github.com/Agoric/agoric-sdk/issues/7825). 


Edit: ah, provisionPool is part of the inter economy bits: v25:zcf-b1-9f877-provisionPool-governor

**Comment by 0xpatrickdev** [comment]
> Still figuring out how to set DEBUG: "SwingSet:ls,SwingSet:vat" for meaningful swingset logs in this env.

Example of this is here: https://github.com/0xpatrickdev/starship-demo/pull/5
